app/server/src/main/scala/org/bitcoins/server/BitcoinSServerMain.scala

```diff
@@ -246,11 +246,12 @@ class BitcoinSServerMain(override val args: Array[String])
       system: ActorSystem): Future[ChainApi] = {
     val blockEC =
       system.dispatchers.lookup(Dispatchers.DefaultBlockingDispatcherId)
-    val chainApi = ChainHandler.fromDatabase(
-      blockHeaderDAO = BlockHeaderDAO()(blockEC, chainAppConfig),
-      CompactFilterHeaderDAO()(blockEC, chainAppConfig),
-      CompactFilterDAO()(blockEC, chainAppConfig))
+
     for {
+      chainApi <- ChainHandlerCached.fromDatabase(
```

Comment:
Sorry, looks like I was wrong. This was supposed to be a regular `ChainHandler` so the wallet and chain routes use the db and it needs to be changed on Node.scala line 72 to be the cached version so we get the optimizations